roachprod: initialize local clusters with correct GEOS location

otan opened the pull request:
Initialize local clusters with --geo-libs set to the correct location
relative to the node's directory. This re-uses the same logic used to
find the CockroachDB binary when in local mode.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52378)
<!-- Reviewable:end -->

otan committed:
roachprod: initialize local clusters with correct GEOS location

Initialize local clusters with --geo-libs set to the correct location
relative to the node's directory. This re-uses the same logic used to
find the CockroachDB binary when in local mode.

Release note: None

otan commented:
going to do this using os.Args[0]
